src/heretic/main.py

```diff
@@ -49,6 +51,106 @@
 )
 
 
+def get_merged_model(model: Model, settings: Settings):
```

Comment:
This should be a method of `Model`, which already has the `Settings` object so this will be much cleaner, and in the semantically correct place.